server/workflow-client.js

```diff
@@ -116,14 +116,14 @@ function uiTransform(item) {
       if (subkey === _payloads) {
         let values = [];
         Object.entries(subvalue).forEach(([subkey, payload]) => {
+          const encoding = Buffer.from(payload.metadata.encoding).toString();
           if (
-            ['json/plain', 'protobuf-json'].includes(payload.metadata.encoding) &&
+            ['json/plain', 'json/protobuf'].includes(encoding) &&
             payload.data
           ) {
-            values = [...values, payload.data.toString('utf8')];
-          }
-          else if (typeof payload.data === 'string'){
-            values = [...values, payload.data.slice(0, 20)];
+            values = [...values, Buffer.from(payload.data).toString()];
+          } else {
+            values = [...values, Buffer.from(payload.data).toString().slice(0, 20)];
```

Comment:
I won't crush if `data` is `null`, right?

Reply:
according to gRPC contract data and encoding would be represented as empty buffer (zero length). Still added checks just in case because of reliance on protobufjs library conversion of proto bytes to javascript Buffer